napi-inl.h

```diff
@@ -3697,10 +3697,26 @@ inline napi_value ObjectWrap<T>::ConstructorCallbackWrapper(
   }
 
   T* instance;
-  napi_value wrapper = details::WrapCallback([&] {
+  napi_value wrapper = details::WrapCallback([&] () -> napi_value {
```

Comment:
```suggestion
  napi_value wrapper = details::WrapCallback([&] {
```

We don't need this change, because it's OK to return `callbackInfo.This()`. After all, when control returns to the engine it will throw anyway, so the actual instance will not be relevant, and there won't be a leak on the native side either.